libs/designsystem/header/src/header.component.ts

```diff
@@ -117,17 +122,12 @@ export class HeaderComponent implements AfterContentInit, OnInit {
       ],
       parent: this.injector,
     });
-
-    if (this.titleMaxLines > 0) {
-      this.fitHeadingConfig = {
-        maxLines: this.titleMaxLines,
-      };
-    }
   }
 
-  ngAfterContentInit(): void {
+  private onPotentialAvatarRendered(): void {
     // If an avatar is present we default to centered layout - unless configured otherwise:
-    if (this.avatar && this.centered === undefined) {
+    const avatarPresent: boolean = this.avatarContainerElement?.nativeElement.children.length > 0;
```

Comment:
Consider checking for visibility instead to better reflect the UI logic for the avatar container:
```suggestion
    const avatarPresent: boolean = this.avatarContainerElement?.nativeElement. checkVisibility();
```